File: app/src/main/java/com/siavash/dualcamera/ApplicationBase.java
package com.siavash.dualcamera;

import android.app.Application;
import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;

import com.squareup.leakcanary.LeakCanary;
import com.squareup.leakcanary.RefWatcher;

/**
 * Created by sia on 9/3/15.
 */
public class ApplicationBase extends Application {
    private static String sAppName;
    private RefWatcher refWatcher;

    public static RefWatcher getRefWatcher(Context context) {
        ApplicationBase appBase = (ApplicationBase) context.getApplicationContext();
        return appBase.refWatcher;
    }

    public static String getAppName() {
        return ApplicationBase.sAppName;
    }

    @Override public void onCreate() {
        super.onCreate();
        refWatcher = LeakCanary.install(this);
        try {
            ApplicationInfo appInfo = getPackageManager().getApplicationInfo(getPackageName(), PackageManager.SIGNATURE_MATCH);
            sAppName = appInfo != null ? (String) getPackageManager().getApplicationLabel(appInfo) : "unknown";
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/kotlin/com/siavash/dualcamera/ApplicationBase.kt
package com.siavash.dualcamera

import android.app.Application
import android.content.pm.PackageManager
import com.crashlytics.android.Crashlytics
import com.crashlytics.android.answers.Answers
import com.crashlytics.android.core.CrashlyticsCore
import io.fabric.sdk.android.Fabric

/**
 * Created by sia on 11/11/15.
 */
class ApplicationBase : Application() {
    companion object ApplicationName {
        lateinit var appName: String
    }

    override fun onCreate() {
        super.onCreate()

        val appInfo = packageManager.getApplicationInfo(packageName, PackageManager.SIGNATURE_MATCH)
        appName = packageManager.getApplicationLabel(appInfo) as String

        val crashlyticsKit = Crashlytics.Builder()
                .core(CrashlyticsCore.Builder().disabled(BuildConfig.DEBUG).build())
                .build()
        Fabric.with(this, crashlyticsKit);
        Fabric.with(this, Answers());
    }
}